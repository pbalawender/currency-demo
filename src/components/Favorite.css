.Favorite {

}

.Favorite .clearAll {
    float: right;
    margin-top: 10px;
}

.Favorite .empty {
    border: 1px dashed #DEDEDE;
}
