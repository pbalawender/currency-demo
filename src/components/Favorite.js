import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import CurrencyList from './CurrencyList';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
import './Favorite.css';

export default class Favorite extends PureComponent {
    static propTypes = {
        favoriteCurrencies: PropTypes.arrayOf(PropTypes.shape({
            code: PropTypes.string.isRequired,
            currency: PropTypes.string,
            mid: PropTypes.number,
            favorite: PropTypes.bool
        })).isRequired,
        toggleFavorite: PropTypes.func.isRequired,
        clearFavorite: PropTypes.func.isRequired
    };

    render() {
        if (!this.props.favoriteCurrencies || this.props.favoriteCurrencies.length === 0) {
            return <h3 className="empty">Start watching currency by selecting <FavoriteBorderIcon/> on the list below </h3>;
        }

        return (
            <div className="Favorite clearfix">
                <CurrencyList currencies={this.props.favoriteCurrencies} toggleFavorite={this.props.toggleFavorite}/>
                <Button className="clearAll" onClick={this.props.clearFavorite}>Clear all</Button>
            </div>
        );
    }
}
