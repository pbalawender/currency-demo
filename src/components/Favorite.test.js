import React from 'react';
import renderer from 'react-test-renderer';
import Favorite from './Favorite';

describe('Favorite component', () => {
    it('renders correctly with empty list', () => {
        const tree = renderer.create(
            <Favorite favoriteCurrencies={[]}
                      toggleFavorite={() => {}}
                      clearFavorite={() => {}} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with non-empty list', () => {
        const mockData = [{
            currency: 'dolar ameryka≈Ñski',
            code: 'USD',
            mid: 3.7759,
            favorite: true
        }];
        const tree = renderer.create(
            <Favorite favoriteCurrencies={mockData}
                      toggleFavorite={() => {}}
                      clearFavorite={() => {}} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
