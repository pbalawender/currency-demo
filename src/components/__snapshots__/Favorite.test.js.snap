// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite component renders correctly with empty list 1`] = `
<h3
  className="empty"
>
  Start watching currency by selecting 
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1"
    color={undefined}
    focusable="false"
    viewBox="0 0 24 24"
  >
    <g>
      <path
        d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
      />
    </g>
  </svg>
   on the list below 
</h3>
`;

exports[`Favorite component renders correctly with non-empty list 1`] = `
<div
  className="Favorite clearfix"
>
  <div
    className="CurrencyList"
  >
    <div
      className="CurrencyDetails header"
    >
      <div
        className="code"
      >
        Code
      </div>
      <div
        className="currency"
      >
        Currency name
      </div>
      <div
        className="mid"
      >
        Rate
      </div>
      <div
        className="favorite"
      >
        Â 
      </div>
    </div>
    <div
      className="CurrencyDetails"
    >
      <div
        className="code"
      />
      <div
        className="currency"
      />
      <div
        className="mid"
      />
      <label
        className="MuiFormControlLabel-root-8 favorite"
      >
        <span
          className="MuiButtonBase-root-26 MuiIconButton-root-20 MuiSwitchBase-root-16 MuiCheckbox-root-11 MuiCheckbox-colorSecondary-15"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-25"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </g>
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-19"
              disabled={undefined}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-29"
          />
        </span>
        <span
          className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-10"
        >
          
        </span>
      </label>
    </div>
  </div>
  <button
    className="MuiButtonBase-root-26 MuiButton-root-60 MuiButton-text-62 MuiButton-flat-65 clearAll"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-61"
    >
      Clear all
    </span>
    <span
      className="MuiTouchRipple-root-29"
    />
  </button>
</div>
`;
